### PR TITLE
Cherry-pick a3p and agd changes for upgrade-14

### PR DESCRIPTION
**Pull request by mhofman**
Staged on top of #8869

## Description

`a3p-integration` and other integration related tests sync from the `master` branch, including a core-eval test for the issue discovered during #8869

This also picks up changes to the `bin/agd` wrapper which had improvements to more reliably rebuild.

Constructed from the following rebase-todo:
```
# Branch Agoric-mfig-tweak-agd
label base-Agoric-mfig-tweak-agd
pick ce7d183cd fix(agd): make script more robust
label Agoric-mfig-tweak-agd
reset base-Agoric-mfig-tweak-agd
merge -C 2f87d77e4 Agoric-mfig-tweak-agd # Merge pull request #8256 from Agoric/mfig-tweak-agd

# From pull request #8001
pick e7ed7bf5a fix(bin-agd): recompile when C++ src changes

# Branch Agoric-mfig-integration-deps
label base-Agoric-mfig-integration-deps
pick e5a80fc8f ci(registry): `set-versions.sh` cwd is already at the target
pick 3bace780d ci(registry): use exact Endo version matches
# pick e0741e50e build(eslint-config): fix `peerDependencies.eslint-plugin-jsdoc`
label Agoric-mfig-integration-deps
reset base-Agoric-mfig-integration-deps
merge -C 01baa2896 Agoric-mfig-integration-deps # Merge pull request #8751 from Agoric/mfig-integration-deps

# Branch Agoric-mfig-build-uses-hashes
label base-Agoric-mfig-build-uses-hashes
pick 456f11c41 build(agd): rely on checksums, not timestamps
pick d712be231 fix(registry.sh): use `build:types` and `clean:types` if they exist
pick a1cfba894 build(golang): remove sources when building Docker image
pick e7fb9f5ab fix(deployment): remove dependency on `HELPER_BINARY`
label Agoric-mfig-build-uses-hashes
reset base-Agoric-mfig-build-uses-hashes
merge -C 386cc4922 Agoric-mfig-build-uses-hashes # Merge pull request #8715 from Agoric/mfig-build-uses-hashes

# From pull request #8866
pick d5484e3ff ci(integration): fix `agd` wrapper and `Dockerfile.sdk`

# Branch Agoric-ta-bump-a3p
label base-Agoric-ta-bump-a3p
pick 160fd0c2c chore(a3p): bump synthetic-chain to 0.0.5-2
pick 5b8663c98 chore(a3p): bump yarn to 4.1.0
pick 1a7d9b342 chore(a3p): bump synthetic-chain to 0.0.5-3
label Agoric-ta-bump-a3p
reset base-Agoric-ta-bump-a3p
merge -C 592f960d2 Agoric-ta-bump-a3p # Merge pull request #8873 from Agoric/ta/bump-a3p

# Pull Request #8892
pick 7e8835ec7 ci: merge integration branch before endo checkout (#8892)

# Branch Agoric-ta-integration-cleanup
label base-Agoric-ta-integration-cleanup
pick a5c64c69f chore: remove obsolete issue template
pick 73a52d415 ci: delete disabled dapp-fungible-faucet test
label Agoric-ta-integration-cleanup
reset base-Agoric-ta-integration-cleanup
merge -C 17e0efe7d Agoric-ta-integration-cleanup # Merge pull request #8899 from Agoric/ta/integration-cleanup

# From pull request #8902
pick 229b36ff1 chore(a3p): bump synthetic-chain to 0.0.6-4

# Branch Agoric-ta-a3p-coreeval
label base-Agoric-ta-a3p-coreeval
pick bab2f8295 docs(a3p): UPGRADE NEEDED case
pick 022abb15c chore(deps): simplify a3p-integration package
# pick f864dd35d test(a3p): restart-vats
label Agoric-ta-a3p-coreeval
reset base-Agoric-ta-a3p-coreeval
merge -C b4094ba2a Agoric-ta-a3p-coreeval # Merge pull request #8801 from Agoric/ta/a3p-coreeval

# Branch Agoric-mhofman-reland-8801
label base-Agoric-mhofman-reland-8801
pick 4ed3581e6 ci: build sdk in upgrade test for potential proposal uses
edit 2a7bcead2 test(a3p): restart vats # rm -rf a3p-integration/proposals/b:restart-vats
label Agoric-mhofman-reland-8801
reset base-Agoric-mhofman-reland-8801
merge -C c839fb4a0 Agoric-mhofman-reland-8801 # Merge pull request #8907 from Agoric/mhofman/reland-8801

# Pull Request #8944
edit 27079f984 refactor(a3p): reusable submission generation function (#8944) # remove buildSubmission restart-vats

# Branch Agoric-mhofman-clarify-a3p
label base-Agoric-mhofman-clarify-a3p
pick b89aa57ca fix(a3p-integration): rebuild SDK image automatically
pick 2798b83db fixup! fix(a3p-integration): rebuild SDK image automatically # Manually authored
pick 2eb9ff4c7 docs: detail structure of a3p-integration
# pick 66ab6c362 docs: add README.md to a3p upgrade proposal
label Agoric-mhofman-clarify-a3p
reset base-Agoric-mhofman-clarify-a3p
merge -C c7efebaff Agoric-mhofman-clarify-a3p # Merge pull request #8886 from Agoric/mhofman/clarify-a3p

# Pull Request #8929
pick bc337b531 feat(a3p-integration): add core eval in test (#8929)

# Pull Request #8940
pick 0c299962f fix(a3p): serial test execution (#8940)

# Branch Agoric-mfig-silence-vat-replay-console
label base-Agoric-mfig-silence-vat-replay-console
pick 5d00dc6dd fix(SwingSet): silence vat console output during replay
pick 3098a8e75 chore(cosmic-swingset): log swingset launched
label Agoric-mfig-silence-vat-replay-console
reset base-Agoric-mfig-silence-vat-replay-console
merge -C a239f8b79 Agoric-mfig-silence-vat-replay-console # Merge pull request #8943 from Agoric/mfig-silence-vat-replay-console

# Branch Agoric-mfig-silence-vat-replay-console
label base-Agoric-mfig-silence-vat-replay-console
pick 5d00dc6dd fix(SwingSet): silence vat console output during replay
pick 3098a8e75 chore(cosmic-swingset): log swingset launched
label Agoric-mfig-silence-vat-replay-console
reset base-Agoric-mfig-silence-vat-replay-console
merge -C a239f8b79 Agoric-mfig-silence-vat-replay-console # Merge pull request #8943 from Agoric/mfig-silence-vat-replay-console
```

### Security Considerations

None

### Scaling Considerations

None

### Documentation Considerations

I don't think the `agd` change warrant documentation change, but could be mentioned in the release notes. The rest is internal

### Testing Considerations

Improvements of the test integration test infra

### Upgrade Considerations

Part of upgrade-14 supporting changes, but not in itself affecting the release.

### CODE REVIEW

**Comment by mhofman** [comment]
@ivanlei reports problems with a missing `tsx` in his environment. Likely a bug of `@agoric/synthetic-chain`. Should probably pick up a fix before merging

File: scripts/generate-a3p-submission.sh
Comment (gibson042) [inline scripts/generate-a3p-submission.sh:17]: This function appears to be a manual introduction of `edit 2a7bcead2 test(a3p): restart vats # remove the restart vats part`, and AFAICT is never called by anything. Did you intend to end the file with an invocation like `buildSubmission "$@"`?

**Review by gibson042 — APPROVED** [review]
Everything lines up except for scripts/generate-a3p-submission.sh (see specific comment below). I feel underqualified to evaluate many of the selections and the corresponding code changes, but they're already on master so a more mechanical review seems acceptable. LGTM modulo confirmation of the shell script.

File: scripts/generate-a3p-submission.sh
Comment (mhofman) [reply scripts/generate-a3p-submission.sh:17]: Right it's actually a change I should land on master first. The `restart-vats` inclusion doesn't make sense in the release branch test

**Comment by mhofman** [comment]
> @ivanlei reports problems with a missing `tsx` in his environment. Likely a bug of `@agoric/synthetic-chain`. Should probably pick up a fix before merging

To close the loop, the issue is documented as https://github.com/Agoric/agoric-3-proposals/issues/110, but as a workaround, we'll rely on global installation for now.

**Pull request merged**
